cfgmgr/portmgr.cpp

```diff
@@ -83,6 +84,47 @@ bool PortMgr::setPortAdminStatus(const string &alias, const bool up)
     return true;
 }
 
+
+bool PortMgr::setPortDHCPMitigationRate(const string &alias, const string &dhcp_rate_limit)
+{
+    // If dhcp_rate_limit is not configured (empty string), do nothing
+    if (dhcp_rate_limit.empty())
+    {
+        SWSS_LOG_DEBUG("DHCP rate limit not configured for port %s, skipping TC configuration", alias.c_str());
+        return true;
+    }
+
+    stringstream cmd;
+    string res, cmd_str;
+    int ret;
+
+    if (dhcp_rate_limit != "0")
+    {
+        int byte_rate = stoi(dhcp_rate_limit) * PACKET_SIZE;
+        cmd << TC_CMD << " qdisc add dev " << shellquote(alias) << " handle ffff: ingress" << " && " \
+            << TC_CMD << " filter add dev " << shellquote(alias) << " protocol ip parent ffff: prio 1 u32 match ip protocol 17 0xff match ip dport 67 0xffff police rate " << to_string(byte_rate) << "bps burst " << to_string(byte_rate) << "b conform-exceed drop";
+        cmd_str = cmd.str();
+        ret = swss::exec(cmd_str, res);
+    }
+    else
+    {
+        // tc qdisc del dev <port_name> handle ffff: ingress
+        cmd << TC_CMD << " qdisc del dev " << shellquote(alias) << " handle ffff: ingress";
+        cmd_str = cmd.str();
+        ret = swss::exec(cmd_str, res);
+    }
+
+    if (!ret)
+    {
+        return true;
+    }
+    else
+    {
+        throw runtime_error(cmd_str + " : " + res);
```

Comment:
where is this exception handled? We don't want portmgr to exit due to an unhandled exception.